kernel/univ.ml

```diff
@@ -418,46 +400,29 @@ struct
     else
       List.Smart.map (fun x -> Expr.successor x) l
 
-  let rec merge_univs l1 l2 =
+  (* Returns the formal universe that is greater than the universes u and v.
+     Used to type the products. *)
+  let rec sup (l1:t) (l2:t) : t =
     match l1, l2 with
     | [], _ -> l2
     | _, [] -> l1
     | h1 :: t1, h2 :: t2 ->
-       let open Expr in
-       (match super h1 h2 with
-        | SuperSame true (* h1 < h2 *) -> merge_univs t1 l2
-        | SuperSame false -> merge_univs l1 t2
-        | SuperDiff c ->
-           if c <= 0 (* h1 < h2 is name order *)
-           then cons h1 (merge_univs t1 l2)
-           else cons h2 (merge_univs l1 t2))
-
-  let sort u =
-    let rec aux a l =
-      match l with
-      | b :: l' ->
-        let open Expr in
-        (match super a b with
-         | SuperSame false -> aux a l'
-         | SuperSame true -> l
-         | SuperDiff c ->
-            if c <= 0 then cons a l
-            else cons b (aux a l'))
-      | [] -> cons a l
-    in
-      List.fold_right (fun a acc -> aux a acc) u []
-
-  (* Returns the formal universe that is greater than the universes u and v.
-     Used to type the products. *)
-  let sup x y = merge_univs x y
+      let cmp = Level.compare (fst h1) (fst h2) in
```

Comment:
Can't you just use List.merge?

Reply:
When we have `u, 0` and `u, 1` we want to keep `u, 1` and forget `u, 0`, I don't think that can be expressed in terms of List.merge